gr-blocks/lib/throttle_impl.cc

```diff
@@ -76,20 +76,18 @@ int throttle_impl::work(int noutput_items,
         }
     }
 
-    // calculate the expected number of samples to have passed through
-    boost::system_time now = boost::get_system_time();
-    boost::int64_t ticks = (now - d_start).ticks();
-    uint64_t expected_samps = uint64_t(d_samps_per_tick * ticks);
+    auto now = std::chrono::steady_clock::now();
+    auto expected_time = d_start + d_sample_period * d_total_samples;
```

Comment:
This looks wrong, can you please check? The expected time should depend on noutput_items, and `d_total_samples` looks like it's always zero.

Reply:
`d_total_samples` is an accumulator, and `noutput_items` is added to it on line 97.

While working on this, I did ponder whether the accumulator should be incremented *before* the sleep rather than after. It might make more sense to use the *last* sample as the time reference rather than the first.

Reply:
I hesitated to change that since I didn't want to change the behaviour of the throttle, but I'd be happy to change the order if it makes sense to others.

Reply:
I agree: Right now, the first call to work is never throttled.
Changing the behaviour doesn't seem to bad. First, this is for 3.9 so we're allowed to change, and second, I can't see how this would break someone's app. Finally, it makes a QA code very easy: You request a rate of say 4Hz, pipe one sample through the block, and make sure the QA takes at least 1/4 second.

Reply:
Oh and yeah forgot my original comment. I missed that accumulator line.

Reply:
OK, I've added a second commit which defers the sleep until after copying samples and accumulating.